protoc_plugin/lib/src/protobuf_field.dart

```diff
@@ -140,6 +140,12 @@ class ProtobufField {
     // for example in package:protobuf/src/protobuf/mixins/well_known.dart.
   }
 
+  bool get isNullable {
+    if (isRepeated) return false;
+    if (isRequired) return false;
+    return descriptor.proto3Optional || baseType.isMessage;
+  }
```

Comment:
Correct me if I am wrong. I am just following this PR because it is a feature I highly request as well. But I think that the `baseType.isMessage` is there because according to the protobuf docs default value for message fields should be langauge specific null value. Or is this handled by the `isRequired` case?

Reply:
That's indeed why I put the `isMessage` @DavidDomkar. But I don't know if it's managed by the `isRequired` actually, I'll test locally.

Reply:
> But I think that the baseType.isMessage is there because according to the protobuf docs default value for message fields should be langauge specific null value.

The "language specific null value" in protobuf.dart is the empty message, which we don't need to change in this PR. If the field is "optional" (whether it's a message or not) we can make it nullable, leaving values of missing non-optional message fields the same.

> Or is this handled by the isRequired case?

I don't think so, `isRequired` just checks the label: https://github.com/google/protobuf.dart/blob/75bc380a6e9601565606a13dee16c5ab2bf5019c/protoc_plugin/lib/src/protobuf_field.dart#L62-L63 if you don't have the "optional" label in the field it will be `false`.

Reply:
I just pushed these changes, I tested locally and it works for me!